File: LeetCode_Java/0053.最大子数组和.java
/*
 * @lc app=leetcode.cn id=53 lang=java
 *
 * [53] 最大子数组和
 */

// @lc code=start
class Solution {
    public int maxSubArray(int[] nums) {
        int res = Integer.MIN_VALUE;
        int t = 0;
        for(int num : nums) {
            t += num;
            res = Math.max(res, t);
            t = t < 0 ? 0 : t;
        }
        return res;
    }
}
// @lc code=end

File: LeetCode_C++/0053.最大子数组和.cpp
/*
 * @lc app=leetcode.cn id=53 lang=cpp
 *
 * [53] 最大子数组和
 */

// @lc code=start
class Solution {
public:
    int maxSubArray(vector<int>& nums) {
        int res = INT_MIN, sum = 0;
        for (int num : nums) {
            sum += num;
            res = max(res, sum);
            sum = max(sum, 0);
        }
        return res;
    }
};
// @lc code=end

